include/exadg/solvers_and_preconditioners/utilities/linear_algebra_utilities.h

```diff
@@ -133,6 +128,40 @@ apply_petsc_operation(
 }
 #endif
 
+
+/**
+ * Certain functionality of e.g. Trilinos is only available in double precision. The aim of this
+ * function is to hide this circumstance from the user code (that uses the template parameter
+ * Number) and to perform Number -> double conversions and then call the double-version of a certain
+ * Trilinos functionality.
+ */
+template<typename Number>
+void
+apply_function_in_double_precision(
+  dealii::LinearAlgebra::distributed::Vector<Number> &                            dst,
+  dealii::LinearAlgebra::distributed::Vector<Number> const &                      src,
+  std::function<void(dealii::LinearAlgebra::distributed::Vector<double> &,
+                     dealii::LinearAlgebra::distributed::Vector<double> const &)> operation)
+{
+  if constexpr(std::is_same_v<Number, double>)
+  {
+    operation(dst, src);
+  }
+  else
+  {
+    // create temporal vectors of type double
+    dealii::LinearAlgebra::distributed::Vector<double> dst_double, src_double;
+    dst_double.reinit(dst, false); // zero entries
```

Comment:
related to comment https://github.com/exadg/exadg/pull/681#discussion_r1680581114

@kronbichler I introduced this free function to avoid code duplication at different places in ExaDG. `reinit(dst, false)` is as in the previous implementation. So my questions from the other comment would be whether we should write `reinit(dst, true)` here?

Reply:
done